src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/JDBCResourceStore.java

```diff
@@ -114,7 +115,8 @@ public JDBCResourceStore(DataSource ds, JDBCResourceStoreProperties config, Reso
     
     @Override
     public Resource get(String path) {
-        if (oldResourceStore != null && ArrayUtils.contains(dir.getConfig().getIgnoreDirs(), path)) {
+        if (oldResourceStore != null && !path.equals(Paths.BASE) &&
+                ArrayUtils.contains(dir.getConfig().getIgnoreDirs(), Paths.names(path).get(0))) {
```

Comment:
This logic checks top-level folders (at the time of writing - data, jdbcstore, workspaces, layergroups, temp, tmp).

Additional logic may be needed, see below.